pkg/rehearse/rehearse.go

```diff
@@ -581,11 +586,7 @@ func setupDependencies(
 // Allow manipulating the speed of time for tests
 var second = time.Second
 
-func ensureImageStreamTags(ctx context.Context, client ctrlruntimeclient.Client, ists apihelper.ImageStreamTagMap, clusterName, namespace string, istImportClient ctrlruntimeclient.Client, log *logrus.Entry) error {
```

Comment:
Good work!
I believe that you find the right place to implement the feature.
However, what is in my mind is to do the mirroring in addition to the existing imports, instead of replacing it which is probably the next card.
I probably did not make this clear in the card and revised it a couple of mins ago.

In general, we might want to do another function `ensureImageInQCI()` where we did the similar thing:
- check if the corresponding image is there or not (and stale or not)
- if not up to date, do an oc-image-mirror

The logic is more or less the same as reconcile function in `ci-images-mirror`.

https://github.com/openshift/ci-tools/blob/09426d1c21edd095bac4308f9473e511a020c7a1/pkg/controller/quay_io_ci_images_distributor/quay_io_ci_images_distributor.go#L132

Reply:
So if I understand you correctly, we want to do the opposite? After the import is created, check if this image exists in quay and mirror it there. Not mirror from quay to the local registry, but from local registry to quay?